packages/next/src/cli/next-dev.ts

```diff
@@ -392,12 +419,175 @@ If you cannot make the changes above, but still want to try out\nNext.js v13 wit
     // this is a temporary solution until we can fix the memory leaks.
     // the logic for the worker killing itself is in `packages/next/server/lib/start-server.ts`
     if (!process.env.__NEXT_DISABLE_MEMORY_WATCHER && cluster.isMaster) {
-      cluster.fork()
-      cluster.on('exit', (worker) => {
-        if (worker.exitedAfterDisconnect) {
-          cluster.fork()
+      let config: NextConfig
+
+      const setupFork = (env?: Parameters<typeof cluster.fork>[0]) => {
+        const startDir = dir
+        let shouldFilter = false
+        cluster.fork({
+          ...env,
+          FORCE_COLOR: '1',
+        })
+
+        // since errors can start being logged from the fork
+        // before we detect the project directory rename
+        // attempt suppressing them long enough to check
+        const filterForkErrors = (chunk: Buffer, fd: 'stdout' | 'stderr') => {
+          const cleanChunk = stripAnsi(chunk + '')
+          if (
+            cleanChunk.match(
+              /(ENOENT|Module build failed|Module not found|Cannot find module)/
+            )
+          ) {
+            if (startDir === dir) {
+              try {
+                // check if start directory is still valid
+                const result = findPagesDir(
+                  startDir,
+                  !!config.experimental?.appDir
+                )
+                shouldFilter = !Boolean(result.pagesDir || result.appDir)
+              } catch (_) {
+                shouldFilter = true
+              }
+            }
+            if (shouldFilter || startDir !== dir) {
+              shouldFilter = true
+              return
+            }
+            process[fd].write(chunk)
+          } else {
+            process[fd].write(chunk)
```

Comment:
Since both the `if` and `else` do `process[fd].write(chunk)`, you can probably move that below and remove the `else`

Reply:
Updated